controller/healthsyncer/healthsyncer.go

```diff
@@ -0,0 +1,133 @@
+package healthsyncer
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	client "github.com/rancher/cluster-agent/client"
+	"github.com/rancher/cluster-agent/controller"
+	clusterv1 "github.com/rancher/types/apis/cluster.cattle.io/v1"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	syncInterval                    = 15 * time.Second
+	ClusterConditionProvisioned     = "Provisioned"
+	ClusterConditionReady           = "Ready"
+	ClusterConditionStatusHealthy   = "True"
+	ClusterConditionStatusUnHealthy = "False"
+)
+
+type HealthSyncer struct {
+	client      *client.Clients
+	clusterName string
+}
+
+func init() {
+	h := &HealthSyncer{}
+	controller.RegisterController(h.GetName(), h)
+}
+
+func (h *HealthSyncer) GetName() string {
+	return "HealthSyncer"
+}
+
+func (h *HealthSyncer) Run(ctx context.Context, clusterName string, client *client.Clients) error {
+	h.clusterName = clusterName
+	h.client = client
+	go h.syncHealth(syncInterval)
+	return nil
+}
+
+func (h *HealthSyncer) syncHealth(syncHealth time.Duration) {
+	for _ = range time.Tick(syncHealth) {
+		err := h.updateClusterHealth()
+		if err != nil {
+			logrus.Info(err)
+		}
+	}
+}
+
+func (h *HealthSyncer) updateClusterHealth() error {
+	cluster, err := h.getCluster()
+	if err != nil {
+		return err
+	}
+	if cluster == nil {
+		logrus.Info("Skip updating cluster health, cluster [%s] deleted", h.clusterName)
+		return nil
+	}
+	if !isProvisioned(cluster) {
+		return fmt.Errorf("Skip updating cluster health - cluster [%s] not provisioned yet", h.clusterName)
+	}
+	cses, err := h.client.ClusterClientV1.ComponentStatuses("").List(metav1.ListOptions{})
+	if err != nil {
+		logrus.Debugf("Error getting componentstatuses for server health %v", err)
+		updateConditionStatus(cluster, ClusterConditionReady, ClusterConditionStatusUnHealthy)
+		return nil
+	}
+	updateConditionStatus(cluster, ClusterConditionReady, ClusterConditionStatusHealthy)
+	logrus.Infof("Cluster [%s] Condition Ready", h.clusterName)
```

Comment:
Do we log `logrus.Infof("Cluster [%s] Condition NotReady", h.clusterName)` anywhere?